### PR TITLE
update weighting for the Aggregate score based on WEO2023

### PR DESCRIPTION
**Pull request by Antoine-Lalechere**
@jdhoffa @MonikaFu 
here are the lines of code to update the weighting for WEO2023 for the Aggregate Score.

Scenario raw data are stored on the following link in Azure but I don't know how to link Azure and GH, so the link used is on my machine for now - can you help me with it? https://portal.azure.com/#view/Microsoft_Azure_FileStorage/FileShareMenuBlade/~/browse/storageAccountId/%2Fsubscriptions%2Ffeef729b-4584-44af-a0f9-4827075512f9%2FresourceGroups%2FRMI-SP-PACTA-PROD%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fpactarawdata/path/scenario-sources/protocol/SMB

### CODE REVIEW

**Comment by MonikaFu** [comment]
thanks @Antoine-Lalechere. Reviewing now. If you want to use a path to azure storage instead of your local path, I don't think that this is possible. We could add a location of the file in a comment maybe. @jdhoffa do you have any input here?

**Comment by jdhoffa** [comment]
Yeah, I would just comment the line out, and add a comment pointing to the relevant file on azure
I guess it's somewhere here:
https://portal.azure.com/#view/Microsoft_Azure_FileStorage/FileShareMenuBlade/~/browse/storageAccountId/%2Fsubscriptions%2Ffeef729b-4584-44af-a0f9-4827075512f9%2FresourceGroups%2FRMI-SP-PACTA-PROD%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fpactarawdata/path/scenario-sources/protocol/SMB

**Comment by jdhoffa** [comment]
More specifically, here:
https://pactarawdata.file.core.windows.net/scenario-sources/weo_2023-20240222/WEO2023 extended data/WEO2023_Extended_Data.xlsx

**Comment by jdhoffa** [comment]
> The last link did not work for me @jdhoffa. Should it work in a browser? My experience with azure is that you can point to a directory but not a specific file. So you need to give a file location in the directory next to the link in the comment.

Hmm indeed, it doesn't work for me either that's odd... that is the URL i get as a result of "copy link", but anyway, the files are here:
`https://portal.azure.com/#view/Microsoft_Azure_FileStorage/FileShareMenuBlade/~/browse/storageAccountId/%2Fsubscriptions%2Ffeef729b-4584-44af-a0f9-4827075512f9%2FresourceGroups%2FRMI-SP-PACTA-PROD%2Fproviders%2FMicrosoft.Storage%2FstorageAccounts%2Fpactarawdata/path/scenario-sources/protocol/SMB`

and the specific file is: `WEO2023_Extended_Data.xlsx`

I suppose so long as those two pieces of information are tracked for posterity, we are good

**Comment by jdhoffa** [comment]
@MonikaFu @Antoine-Lalechere I leave it to you two to decide exactly how you want to achieve that. 
Agreed that for this project we shouldn't overthink where the data is stored, but probably in the long-term having it in `workflow.*` would make sense.

File: data-raw/remaining_carbon_budgets.R [hunk 1]
```diff
@@ -1,15 +1,55 @@
...
+# The specific file is 'weo_2023-20240222/WEO2023 extended data/WEO2023_Extended_Data.xlsx'
+# Please download the file and replace the dummy path in variable 'file_scenario_emissions_data'
+# to the path on your computer before running the code.
+file_scenario_emissions_data <- "PATH/TO/WEO2023/SCENARIO/EXTENDED/DATA"
+carbon_emissions <- read_xlsx(file_scenario_emissions_data,
+                              sheet = "World CO2 Emissions",
+                              range = "J7:AP45") %>%
+  select(-c("2022...5", "2030...6", "2035...7", "2040...8", "2045...9", "2050...10", "...11", "2030...12", "2050...13",
+            "2022...15", "2030...16", "2035...17", "2040...18", "2045...19", "2050...20", "...21", "2030...22", "2050...23", "...24",
+            "...31", "2030...32", "2050...33", "...14")) %>%
```
Comment (jdhoffa) [inline data-raw/remaining_carbon_budgets.R:13]: NIT: Probably better just to read in the `range` that you want explicitly in the `read_xlsx` function? Also probably good name the columns in a vector on read as well

File: data-raw/remaining_carbon_budgets.R [hunk 2]
```diff
@@ -1,15 +1,55 @@
...
+    scenario_source = "WEO2023",
+    unit = "Mt CO2 (2022)"
   )
+
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
+carbon_emissions <- carbon_emissions[-5,]
```
Comment (jdhoffa) [inline data-raw/remaining_carbon_budgets.R:34]: There's probably a more readable/ interpretable way to do this?

File: data-raw/remaining_carbon_budgets.R [hunk 3]
```diff
@@ -1,15 +1,55 @@
...
+
+
+remaining_carbon_budgets <- carbon_emissions %>%
+  mutate(
+    remaining_carbon_budget = emissions_2022 * 4 + emissions_2030 * 4, # remaining carbon dbusget is the interpolized carbon budget until 2030 for a sector
```
Comment (jdhoffa) [inline data-raw/remaining_carbon_budgets.R:39]: why do we multiply by 4? Please add a comment to explain what is going on there.

**Review by jdhoffa — CHANGES_REQUESTED** [review]

File: data-raw/remaining_carbon_budgets.R (hunk 3 above)
Comment (MonikaFu) [reply data-raw/remaining_carbon_budgets.R:39]: @Antoine-Lalechere ?

File: data-raw/remaining_carbon_budgets.R (hunk 2 above)
Comment (Antoine-Lalechere) [reply data-raw/remaining_carbon_budgets.R:34]: image
The issue is that those lines get the same labels

Comment (MonikaFu) [reply data-raw/remaining_carbon_budgets.R]: I fixed this

File: data-raw/remaining_carbon_budgets.R (hunk 3 above)
Comment (MonikaFu) [reply data-raw/remaining_carbon_budgets.R:30]: @Antoine-Lalechere do we include both 2022 and 2030 in the calculation? Or we start from 2023. Do I understand correctly then that we assume emissions assigned to years 2023-2026 are assumed to be the same as those in 2022 and those for 2027-2030 are assumed to be equal to 2030?

Comment (Antoine-Lalechere) [reply data-raw/remaining_carbon_budgets.R:30]: We do an interpolation of carbon budget (so we have a line for carbon budget by sector between 2022 and 2030)
Then 2023 + 2029 carbon budget sums together and are equal to 2022 + 2030 carbon budget
same for 2024 + 2028 etc...
I think I forgot a year, it should be 4.5 and not 4 in the formula

Comment (MonikaFu) [reply data-raw/remaining_carbon_budgets.R:30]: Ok, I think I understand. In the end after I run some calculations it seems that the sum of linearly interpolated points is the same as what you are doing here. The only unresolved question is if we should be using 2022 or not? Should we? Aren't we using data from 2023 to 2030 for the calculation?>

Comment (Antoine-Lalechere) [reply data-raw/remaining_carbon_budgets.R:30]: The data we get is from 2022 so we have to consider it


Formula should actually be:

`remaining_carbon_budget = emissions_2022 * 4.5 + emissions_2030 * 4.5,`

This won't change the results as we take then percentage

**Review by jdhoffa — APPROVED** [review]

**Review by MonikaFu — APPROVED** [review]

**Pull request merged**